model/task/task_test.go

```diff
@@ -1855,6 +1855,56 @@ func TestUnattainableSchedulableHostTasksQuery(t *testing.T) {
 	assert.Len(schedulableTasks, 2)
 }
 
+func TestUnattainableScheduledContainerTasksQuery(t *testing.T) {
+	assert := assert.New(t)
+	assert.NoError(db.ClearCollections(Collection))
+	tasks := []Task{
+		{
+			Id:                "t0",
+			Activated:         true,
+			Status:            evergreen.TaskUndispatched,
+			ExecutionPlatform: ExecutionPlatformContainer,
+			Priority:          0,
+		},
+		{
+			Id:                "t1",
+			Activated:         true,
+			Status:            evergreen.TaskUndispatched,
+			ExecutionPlatform: ExecutionPlatformContainer,
+			Priority:          -1,
+		},
+		{
+			Id:                "t2",
+			Activated:         true,
+			Status:            evergreen.TaskUndispatched,
+			ExecutionPlatform: ExecutionPlatformContainer,
+			Priority:          2,
+		},
+		{
+			Id:                "t3",
+			Activated:         false,
+			Status:            evergreen.TaskUndispatched,
+			ExecutionPlatform: ExecutionPlatformContainer,
+			Priority:          0,
+		},
+		{
+			Id:                "t4",
+			Activated:         false,
+			Status:            evergreen.TaskUndispatched,
+			ExecutionPlatform: ExecutionPlatformHost,
+			Priority:          0,
+		},
+	}
+	for _, task := range tasks {
+		assert.NoError(task.Insert())
+	}
+
+	q := db.Query(ScheduledContainerTasksQuery())
+	schedulableTasks, err := FindAll(q)
+	assert.NoError(err)
+	assert.Len(schedulableTasks, 2)
```

Comment:
I would test this query more thoroughly since it has multiple checks. Usually when I write unit tests, I consider each clause in the query the equivalent of an "if" statment and to achieve good case coverage, it's important to verify that each individual "special case" in the definition of a "schedulable" task behaves as expected. For example, all the tasks tested in this have "undispatched" status, but it's worth verifying that a task that's lacking the undispatched status is filtered out of the query.

It's also important to test that the tasks you get back from each test are the ones you expect, not just the count (e.g. you could check that task IDs were for the tasks you thought this query would return).

When I have to test several special cases and expected behaviors, I usually do [something like this](https://github.com/evergreen-ci/evergreen/blob/2406d863f1422c46b46ebd4abd2e9f19a12cad74/model/pod/db_test.go#L14-L89).